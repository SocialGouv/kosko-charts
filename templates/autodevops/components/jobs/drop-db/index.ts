import { dropDbJob } from "@socialgouv/kosko-charts/components/azure-pg/drop-db.job";
import { getDevDatabaseParameters } from "@socialgouv/kosko-charts/components/azure-pg/params";
import environments from "@socialgouv/kosko-charts/environments";

const ciEnv = environments(process.env);

const pgParams = getDevDatabaseParameters({ suffix: ciEnv.environment });

export default dropDbJob(pgParams);
