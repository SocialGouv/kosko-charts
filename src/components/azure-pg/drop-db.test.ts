import { dropDbJob } from "./drop-db.job";

process.env.CI_COMMIT_SHORT_SHA = "b123a99";

test("should customize secret and extensions when running drop-db", () => {
  expect(
    dropDbJob({
      database: "some-db",
      user: "some-user",
      secretRefName: "some-secret",
    })
  ).toMatchSnapshot();
});

test("should use defaults when running drop-db", () => {
  expect(
    dropDbJob({
      database: "some-db",
      user: "some-user",
    })
  ).toMatchSnapshot();
});
