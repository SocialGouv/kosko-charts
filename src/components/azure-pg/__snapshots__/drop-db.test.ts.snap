// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should customize secret and extensions when running drop-db 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {
    "name": "drop-azure-db-b123a99",
  },
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "metadata": Object {},
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "drop-db-user",
            ],
            "env": Array [
              Object {
                "name": "DROP_DATABASE",
                "value": "some-db",
              },
              Object {
                "name": "DROP_USER",
                "value": "some-user",
              },
            ],
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "some-secret",
                },
              },
            ],
            "image": "registry.gitlab.factory.social.gouv.fr/socialgouv/docker/azure-db:2.6.1",
            "imagePullPolicy": "IfNotPresent",
            "name": "drop-db-user",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "256Mi",
              },
              "requests": Object {
                "cpu": "100m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;

exports[`should use defaults when running drop-db 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {
    "name": "drop-azure-db-b123a99",
  },
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "metadata": Object {},
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "drop-db-user",
            ],
            "env": Array [
              Object {
                "name": "DROP_DATABASE",
                "value": "some-db",
              },
              Object {
                "name": "DROP_USER",
                "value": "some-user",
              },
            ],
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "azure-pg-admin-user",
                },
              },
            ],
            "image": "registry.gitlab.factory.social.gouv.fr/socialgouv/docker/azure-db:2.6.1",
            "imagePullPolicy": "IfNotPresent",
            "name": "drop-db-user",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "256Mi",
              },
              "requests": Object {
                "cpu": "100m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;
